wagtail/contrib/postgres_search/backend.py

```diff
@@ -226,36 +225,41 @@ def get_search_field(self, field_lookup, fields=None):
                 return self.get_search_field(sub_field_name, field.fields)
 
     # TODO: Find a way to use the term boosting.
-    def check_boost(self, query):
-        if query.boost != 1:
+    def check_boost(self, query, boost=1.0):
+        if query.boost * boost != 1.0:
             warn('PostgreSQL search backend '
                  'does not support term boosting for now.')
 
-    def build_database_query(self, query=None, config=None):
+    def build_database_query(self, query=None, config=None, boost=1.0):
         if query is None:
             query = self.query
 
-        if isinstance(query, SearchQueryShortcut):
-            return self.build_database_query(query.get_equivalent(), config)
-        if isinstance(query, Prefix):
-            self.check_boost(query)
-            self.is_autocomplete = True
-            return PostgresSearchAutocomplete(unidecode(query.prefix),
-                                              config=config)
-        if isinstance(query, Term):
-            self.check_boost(query)
-            return PostgresSearchQuery(unidecode(query.term), config=config)
+        if isinstance(query, PlainText):
+            self.check_boost(query, boost=boost)
+
+            operator = {
+                'and': AND,
+                'or': OR,
+            }[query.operator]
```

Comment:
That dictionary could be moved as a class attribute for performance and consistency with `DEFAULT_OPERATOR`.